### PR TITLE
chore: add verify-docker-build.sh

### LINKED ISSUE #86: Ensure docker build is not broken with a verify script
/kind bug

**What steps did you take and what happened:**
The docker build is broken! It should have a verify script.

### PR DESCRIPTION
**Pull request by dennisme**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #86 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

File: hack/verify-docker-build.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+set -o nounset
+set -o pipefail
+
+# shellcheck source=/dev/null
+source "$(dirname "$0")/utils.sh"
+
+# check if manager docker image builds
+cd_root_path
+
+export GO111MODULE=on
```
Comment (chuckha) [inline hack/verify-docker-build.sh]: i believe we can skip setting this and running go mod download, the docker build should do that for us

Comment (dennisme) [reply hack/verify-docker-build.sh]: ack

**Comment by chuckha** [comment]
ahhh ok, so this does not "just work" but will require a minor change to test-infra. We need to enable docker on our prow image.

Basically needs the stuff the docker provider has: https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-sigs/cluster-api-provider-docker/cluster-api-provider-docker-presubmits.yaml#L6-L19

in our own presubmits yaml: https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-sigs/cluster-api-boostrap-provider-kubeadm/cluster-api-bootstrap-provider-kubeadm-presubmits.yaml

I'm happy to take those over, but if you'd like to get the commits, I can lgtm over there and tag the right people for you

**Comment by chuckha** [comment]
also please squash the commit down to 1 commit (and i believe you have to remove any usernames from the commit message)

**Comment by dennisme** [comment]
@chuckha, yep ill squash. Ill do the test-infra stuff too. Thanks.

**Comment by dennisme** [comment]
@chuckha ok squashed to 1 commit and opened https://github.com/kubernetes/test-infra/pull/13811.

**Comment by chuckha** [comment]
/approve
/lgtm

Thank you for this!

**Pull request merged**